### PR TITLE
Add More Tests for Kubernetes Sals

### PR DESCRIPTION
**Pull request by MElborolossy**
### Description

Add more tests for Kubernetes sals

### Changes

- [x] get_helm_chart_user_values
- [x] upgrade_release --> Wait for modifications in the function

### Related Issues

#2140

### Checklist

- [x] [Pre-commit hook is installed](https://github.com/threefoldtech/js-ng#pre-commit) to do formatting checks before committing code...etc
- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

**Review by waleedhammam — APPROVED** [review]

File: tests/sals/kubernetes/test_kubernetes.py [hunk 1]
```diff
@@ -136,3 +137,86 @@ def test_06_execute_native_cmd(self):
         self.info("Check if command executed correctly")
         self.assertEqual(cmd_out.count("master"), 1)
         self.assertGreaterEqual(cmd_out.count("Ready"), 1)
+
+    def test_07_get_helm_chart_user_values(self):
+        """Test case for get the custom user values for helm chart
```
Comment (AhmedHanafy725) [inline tests/sals/kubernetes/test_kubernetes.py:142]:
```suggestion
        """Test case for getting the custom user values for helm chart
```

File: tests/sals/kubernetes/test_kubernetes.py [hunk 2]
```diff
@@ -136,3 +137,86 @@ def test_06_execute_native_cmd(self):
...
+        self.kube_manager.install_chart(release_name, release_chart, extra_config=input_user_values)
+        self.info("Get user values from get_helm_chart_user_values function")
+        func_user_values_str = self.kube_manager.get_helm_chart_user_values(release_name)
+        # Modify returned value to be comparable
+        func_user_values_str = func_user_values_str.replace("{", "").replace("}", "").replace('"', "").replace("\n", "")
+        self.info("Check if get user input work correctly")
+        self.assertEqual(input_user_values_str, func_user_values_str)
+
+    def test_08_upgrade_release(self):
+        """Test case for upgrade helm release
```
Comment (AhmedHanafy725) [inline tests/sals/kubernetes/test_kubernetes.py:170]:
```suggestion
        """Test case for upgrading helm release.
```

File: tests/sals/kubernetes/test_kubernetes.py [hunk 3]
```diff
@@ -136,3 +137,86 @@ def test_06_execute_native_cmd(self):
...
+        release_name = "testupgraderelease"
+        release_chart = "bitnami/etcd"
+        self.kube_manager.install_chart(release_name, release_chart)
+        self.info("Upgrade Release")
+        upgrade_status = self.kube_manager.upgrade_release(release_name, release_chart)
+        self.info("Check if release upgraded")
+        self.assertTrue(upgrade_status.find(f'Release "{release_name}" has been upgraded. Happy Helming!') != -1)
+
+    def test_09_upgrade_release_with_yaml_config(self):
+        """Test case for upgrade helm release with given yaml config
```
Comment (AhmedHanafy725) [inline tests/sals/kubernetes/test_kubernetes.py:193]:
```suggestion
        """Test case for upgrading helm release with given yaml config.
```

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
we need to delete deployed solution in `tearDown`.

**Comment by MElborolossy** [comment]
The whole vdc will deleted

**Comment by AhmedHanafy725** [comment]
> The whole vdc will deleted

But it might run out of resources?

**Comment by MElborolossy** [comment]
> > The whole vdc will deleted
> 
> 
> 
> But it might run out of resources? 

If it is a new vdc, it will not

**Comment by MElborolossy** [comment]
After last changes, All tests passed
image

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**